packages/orama/src/trees/bkd.ts

```diff
@@ -0,0 +1,280 @@
+export interface Point {
+  x: number
+  y: number
+}
+
+export interface Node {
+  point: Point
+  left?: Node
+  right?: Node
+}
+
+type MaybeNode = Node | undefined
+
+interface InsertTask {
+  points: Point[]
+  depth: number
+  parent?: Node
+  direction?: 'left' | 'right'
+}
+
+interface SearchTask {
+  node: MaybeNode
+  depth: number
+}
+
+const K = 2 // 2D points
+const EARTH_RADIUS = 6371000 // meters
+
+export function create (points: Point[]): MaybeNode {
+  if (points.length === 0) {
+    return undefined
+  }
+
+  const stack: InsertTask[] = [{
+    points,
+    depth: 0
+  }]
```

Comment:
I suggest you initialize all the properties:

```suggestion
  const stack: InsertTask[] = [{
    points,
    depth: 0,
    parent: undefined,
    direction: undefined,
  }]
```